mmf/trainers/lightning_trainer.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import logging
+import math
+
+import omegaconf
+from mmf.common import typings as mmf_typings
+from mmf.common.registry import registry
+from mmf.datasets.lightning_datamodule import LightningDataModule
+from mmf.modules.metrics import Metrics
+from mmf.trainers.base_trainer import BaseTrainer
+from mmf.trainers.lightning_core.loop_callback import LightningLoopCallback
+from mmf.utils.build import build_model
+from mmf.utils.general import get_max_updates, print_model_parameters
+from pytorch_lightning import Trainer
+
+
+logger = logging.getLogger(__name__)
+
+
+@registry.register_trainer("lightning")
+class LightningTrainer(BaseTrainer):
+    def __init__(self, config: mmf_typings.DictConfig):
+        super().__init__(config)
+        self.trainer = None
+
+    def load(self):
+        super().load()
+        self.trainer_config = self.config.trainer.params
+        self._calculate_max_updates()
+        self._load_trainer()
+
+    def _load_trainer(self):
+        lightning_params = self.trainer_config
+        self.trainer = Trainer(
+            logger=False,
+            gpus=lightning_params.gpus,
```

Comment:
You should pass ligthning_params as `**lightning_params` so that if something comes in tomorrow, it can be easily overridden in child configs.